src/test/java/runge/memetests/MemeControllerTest.java

```diff
@@ -0,0 +1,120 @@
+package runge.memetests;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.plugins.RxJavaPlugins;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import org.junit.jupiter.api.Test;
+import runge.meme.*;
+
+import javax.swing.*;
+
+import java.awt.*;
+
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.verify;
+
+public class MemeControllerTest
+{
+
+    static {
+        //makes it so that our service returns right away
+        RxJavaPlugins.setIoSchedulerHandler(scheduler -> Schedulers.trampoline());
+        RxJavaPlugins.setNewThreadSchedulerHandler(scheduler -> Schedulers.trampoline());
```

Comment:
This should be
`RxSwingPlugins.setOnEdtScheduler(scheduler -> Schedulers.trampoline());`